Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -63,18 +63,22 @@ def check_for_prerelease_xcode
       def check_xcode_up_to_date
         return unless MacOS::Xcode.installed? && MacOS::Xcode.outdated?
 
+        message = <<-EOS.undent
+          Your Xcode (#{MacOS::Xcode.version}) is outdated
+          Please update to Xcode #{MacOS::Xcode.latest_version}.
+          #{MacOS::Xcode.update_instructions}
+        EOS
+
         if OS::Mac.prerelease?
-          xcode_select_nudge = <<-EOS.undent
+          current_path = Utils.popen_read("/usr/bin/xcode-select", "-p")
+          message += <<-EOS.undent
             If #{MacOS::Xcode.latest_version} is installed, you may need to:
-              sudo xcode-select --switch /path/to/Xcode-beta.app
+              sudo xcode-select --switch /path/to/valid/Xcode.app
```

Comment:
I reckon it's worth just making this `/Applications` as that's the default location for copy-pasting and anyone else will figure out it needs to be different.